packages/react/src/components/header/components/navigationLanguageSelector/NavigationLanguageSelector.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { PropsWithChildren, useEffect } from 'react';
+
+import {
+  LanguageOption,
+  useActiveLanguage,
+  useSetAvailableLanguages,
+  useSetLanguage,
+} from '../../../../context/languageContext';
+import classNames from '../../../../utils/classNames';
+import { withDefaultPrevented } from '../../../../utils/useCallback';
+import { HeaderActionBarItemWithDropdown } from '../headerActionBarItem';
+import { IconAngleDown, IconAngleUp, IconGlobe } from '../../../../icons';
+import classes from './NavigationLanguageSelector.module.scss';
+import { useHeaderContext } from '../../HeaderContext';
+
+export type LanguageSelectorProps = PropsWithChildren<{
+  /**
+   * Array of languages as LanguageOption objects.
+   */
+  languages: LanguageOption[];
+}>;
+
+const LanguageButton = ({ language }) => {
+  const activeLanguage = useActiveLanguage();
+  const setLanguage = useSetLanguage();
+  const className = classNames(classes.item, { [classes.activeItem]: activeLanguage === language.value });
+  const selectLanguage = withDefaultPrevented(() => setLanguage(language.value));
+
+  return (
+    <button key={language.value} onClick={selectLanguage} type="button" className={className}>
+      <span>{language.label}</span>
+    </button>
+  );
```

Comment:
While trying to test if [element ids are still duplicated](https://helsinkisolutionoffice.atlassian.net/browse/HDS-871), I noticed there is no prop for ids or data-testIds. 

Only text content separate the language option elements, which makes for example robot testing harder. 

VoiceOver seems to pronounce the options correctly on mac/Chrome, but the older version (Navigation component) has lang-attribute in each option, so voiceOver knows how to pronounce them and robot tests can find correct element to click.
  
The button should probably have lang-attribute?